src/library.js

```diff
@@ -4734,30 +4738,55 @@ LibraryManager.library = {
   // JavaScript frames, so we have to rely on PC values. Therefore, we must be able to unwind from
   // a PC value that may no longer be on the execution stack, and so we are forced to cache the
   // entire call stack.
-  emscripten_stack_snapshot__deps: ['emscripten_get_callstack_js', 'emscripten_generate_pc', '$UNWIND_CACHE'],
+  emscripten_stack_snapshot__deps: ['emscripten_get_callstack_js', 'emscripten_generate_pc',
+                                    '$UNWIND_CACHE', '_emscripten_save_in_unwind_cache'],
   emscripten_stack_snapshot: function () {
-    var callstack = _emscripten_get_callstack_js(0).split('\n');
+    var callstack = new Error().stack.split('\n');
```

Comment:
get_callstack_js does some fixing up of the stack trace. do we not need that to work around browser differences? or do we do the same workarounds here too (which is maybe not great if so)?

Reply:
We are working around browser differences here by having separate regexes for v8 and spidermonkey. We used to have that anyways due to minor differences in how wasm stack frames are formatted in the two engines, despite `emscripten_get_callstack_js`'s attempt at unifying it. We might as well cut the middleman and not parse the results twice.